workflow/lib/shared/initialize_paramsearch.py

```diff
@@ -0,0 +1,173 @@
+"""Initialize parameter searches."""
+
+from snakemake.utils import Paramspace
+import pandas as pd
+from itertools import product
+from pathlib import Path
+
+from lib.shared.file_utils import get_filename
+
+def initialize_segment_sbs_paramsearch(config):
+    """Initialize parameter search for sbs segmentation.
+
+    Args:
+        config (dict): Configuration dictionary.
+
+    Returns:
+        dict: Updated configuration dictionary.
+        list: Nuclei diameters.
+        list: Cell diameters.
+        list: Flow thresholds.
+        list: Cell probabilities.
+        dict: Output patterns.
+
+    """
+
+    if config['sbs_process'].get('mode') != 'segment_sbs_paramsearch':
+        return config, None, None, None, None
+        
+    if 'paramsearch' not in config['sbs_process']:
+        base_nuclei = config['sbs_process']['nuclei_diameter']
+        base_cell = config['sbs_process']['cell_diameter']
+        nuclei_diameters = [base_nuclei - 2, base_nuclei, base_nuclei + 2]
+        cell_diameters = [base_cell - 2, base_cell, base_cell + 2]
+        flow_thresholds = [0.2, 0.4, 0.6]
+        cell_probs = [-4, -2, 0, 2, 4]
+    else:
+        nuclei_diameters = config['sbs_process']['paramsearch']['nuclei_diameter']
+        cell_diameters = config['sbs_process']['paramsearch']['cell_diameter']
+        flow_thresholds = config['sbs_process']['paramsearch']['flow_threshold']
+        cell_probs = config['sbs_process']['paramsearch']['cellprob_threshold']
+    
+    SBS_PROCESS_FP = Path(config['all']['root_fp']) / 'sbs_process'
+
+    # Create the output pattern using {wildcards} in the info_type
+    return config, nuclei_diameters, cell_diameters, flow_thresholds, cell_probs, {
+        "segment_sbs_paramsearch": [
+            SBS_PROCESS_FP / "paramsearch" / "images" / get_filename(
+                {"well": "{well}", "tile": "{tile}"},
+                f"paramsearch_nd{{nuclei_diameter}}_cd{{cell_diameter}}_ft{{flow_threshold}}_cp{{cellprob_threshold}}_{suffix}",
+                "tiff"
+            )
+            for suffix in ["nuclei", "cells"]
+        ] + [
+            SBS_PROCESS_FP / "paramsearch" / "tsvs" / get_filename(
+                {"well": "{well}", "tile": "{tile}"},
+                "paramsearch_nd{nuclei_diameter}_cd{cell_diameter}_ft{flow_threshold}_cp{cellprob_threshold}_segmentation_stats",
+                "tsv"
+            )
+        ],
```

Comment:
This might be a good time to change the get_filename and parse_filename methods to take other params. I get worried about adding names without standardization because long term it starts adding up and we run into problems with understanding what names mean or where they come from. Maybe we should discuss this in our weekly sync meeting?

Reply:
Hm, not sure if we need to do this. Doesn't really happen outside the paramsearch?